.github/scripts/generate_ci_workflows.py

```diff
@@ -155,12 +159,21 @@ def assert_valid(self) -> None:
         if self.arch == 'windows':
             assert self.test_runner_type in WINDOWS_RUNNERS, err_message
 
+    def generate_workflow_name(self, workflow: 'CIWorkflow') -> str:
+        workflow_name = workflow.build_environment
+        if not workflow.on_pull_request:
+            workflow_name = "master-only-" + workflow_name
+        if workflow.is_scheduled:
+            workflow_name = "periodic-" + workflow_name
+        return workflow_name
+
     def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
-        output_file_path = GITHUB_DIR / f"workflows/generated-{workflow.build_environment}.yml"
+        workflow.ci_workflow_name = self.generate_workflow_name(workflow)
```

Comment:
nit: This is an immutable property of the `CIWorkflow`, it seems like this should move to `__post_init__` instead